plugins/filter/merge.py

```diff
@@ -70,11 +71,65 @@ def merge_with_app(base, new):
     return result
 
 
+def merge_with_database(base, database, app_name, customer, environment):
+    """
+        Merge a database information with a database structure already existent.
+        If database already exist the new one is ignored.
+    """
+
+    if not isinstance(base, dict) or not isinstance(database, dict):
+        raise AnsibleFilterError("input database is empty")
+
+    if "engine" not in database:
+        raise AnsibleFilterError("input database should define an 'engine' key")
+
+    if "release" not in database:
+        raise AnsibleFilterError("input database should define a 'release' key")
+
+    result = deepcopy(base)
+
+    database_name = "_".join([environment[0], customer, app_name])
+    new_database = {
+        "application": app_name,
+        "password": random_password(),
+        "name": database_name,
+        "user": database_name,
+    }
+
+    engine = database.get("engine")
+
+    if engine not in result:
+        result[database.get("engine")] = [{
+            "release": database.get("release"),
+            "databases": [new_database]
+        }]
+
+        return result
+
+    for defined_engine in result[engine]:
+        if defined_engine.get("release", None) == database.get("release"):
+            for defined_database in defined_engine.get("databases"):
+                if defined_database.get('application') == app_name:
+                    break
+            else:
+                defined_engine.get("databases").append(new_database)
+        else:
+            result[engine].append({
+                "release": database.get("release"),
+                "databases": [new_database]
+            })
```

Comment:
Even if this block is quite self-explanatory, I think you must add comments on conditions to explain every step. Your future you (or every contributor) will thank you for this.